### PR TITLE
Adding Images as Media via the feed: Brute Force

### PR DESCRIPTION
**Pull request by judaw**
First pass/WIP

### CODE REVIEW

File: src/Form/LocalistEntityForm.php [hunk 1]
```diff
@@ -240,13 +240,13 @@ public function form(array $form, FormStateInterface $form_state) {
       '#required' => FALSE,
     ];
 
-    $form['localist_image_field_name'] = [
+    $form['localist_media_field_name'] = [
```
Comment (alisonjo315) [inline src/Form/LocalistEntityForm.php:243]: Woah, it took me a WHILE to figure out that we have this field in here twice haha, I was like, "he's replacing it, but also it's still there, what is going onnnnnnnnn" -- glad that this work will have the side benefit of removing the duplicated form field declaration! 🎉

File: src/LocalistProcessor.php [hunk 2]
```diff
@@ -166,6 +174,30 @@ private function create_file_and_array($url) {
...
+    $media_id = $this->find_media_for_file($image_array);
+    if($media_id) {
+      return $media_id;
+    }
+
+    $media = Media::create([
+      'bundle'           => 'image',
+      'uid'              => 1,
+      'field_media_image' => $image_array,
+    ]);
```
Comment (alisonjo315) [inline src/LocalistProcessor.php:187]: Note to add to media field help/description text (and module README I suppose) that the importer expects there to be an image media bundle called "image" with an image file field called "field_media_image".

(Could the be another form field...?  Suuuure...?  But I think it's okay like this, these are pretty standard bundle/field names.  We can always add another form field to control this setting later, if it becomes a problem.)

File: src/Form/LocalistEntityForm.php [hunk 3]
```diff
@@ -240,13 +240,13 @@ public function form(array $form, FormStateInterface $form_state) {
...
       '#type' => 'textfield',
-      '#title' => $this->t('Localist event image field'),
-      '#default_value' => $localist_pull->localist_image_field_name,
+      '#title' => $this->t('Localist event media field'),
+      '#default_value' => $localist_pull->localist_media_field_name,
       '#size' => 20,
       '#maxlength' => 255,
-      '#description' => $this->t('Mapping: field machine name for the localist image'),
+      '#description' => $this->t('Mapping: field machine name for the localist media'),
```
Comment (alisonjo315) [inline src/Form/LocalistEntityForm.php:249]: Woah, it took me WAY too long to figure out that we have this field in here twice haha, I was like, "he's replacing it, but also it's still there, what is going onnnnnnnnn" -- glad that this work will have the side benefit of removing the duplicated form field declaration! 🎉

**Comment by alisonjo315** [comment]
Notes from chatting with Bill yesterday:
* Recommend changing the target media field to `field_image` (a field that will already exist on most sites, used by news and spotlights -- will make for simpler twig templates, among other things)
* I think we should take the update script out of the repo, put it into a wiki page or something on the repo, and point to the script and how to use it in the release notes
  - keep in mind it's not going to be totally plug-and-play anyway, so it feels like no biggie to not have this in the repo -- and also, I prefer that to having the update script in the repo!

And, idea from Irina at today's devs meeting:
* Along with the update script, provide some config YML files people can either import as-is, or use as models for the config changes they'll need to make before using the update script.

**Comment by alisonjo315** [comment]
"Where to attach the files" -- brainstorming:
- Attach to this PR (on the PR description field)
- Attach to release notes ([example release notes](https://github.com/CU-CommunityApps/CD_cwd_events_localist_pull/releases))
- Create a [wiki page](https://github.com/CU-CommunityApps/CD_cwd_events_localist_pull/wiki)

And, where to put the usage instructions:
- PR description
- Release notes
- Wiki page

The usage instructions and files don't have to be in the same place -- they can be, but they don't have to be.  We'll certainly want to put SOME guidance in the release notes, but we could link to the PR description or the wiki page for full details.  Or it could all go in the release notes.

----

P.S. We can't actually attach YML or PHP files directly, we'll have to archive them into ZIP or GZ files.

File: src/LocalistProcessor.php [hunk 4]
```diff
@@ -166,6 +174,30 @@ private function create_file_and_array($url) {
...
+    $media->setName($media_name)->setPublished(TRUE)->save();
```
Comment (alisonjo315) [inline src/LocalistProcessor.php:188]: Working on the news importer project, using this as an example -- code hints are telling me `setPublished()` does not accept arguments. (_[Relevant source code](https://git.drupalcode.org/project/drupal/-/blob/86d449bc46dbb838c4c51277689d964f58c20a41/core/lib/Drupal/Core/Entity/EntityPublishedTrait.php#L51-59)_)

Recommend:
```suggestion
    $media->setName($media_name)->setPublished()->save();
```

**Comment by judaw** [comment]
Closing for #48

**Pull request closed without merging**